### DISCUSSION TITLE
Settings for large data to get duplicate records cluster together

### DISCUSSION THREAD

**Discussion by abhi12bh** [moj-analytical-services/splink#1627] [General]
My requirement is to match records based on the condition that if two records have the same Postal Code and Province, and their Addresses exhibit a good level of fuzzy matching, then they should be considered a match. However, when I use the code below, it doesn't yield the expected results. Instead, it assigns the same cluster to strings that share the same Postal Code and Province, even when their Addresses differ significantly. My file having 10k records

<img width="528" alt="image" src="https://github.com/moj-analytical-services/splink/assets/30687917/eada140f-f940-4ea2-980d-c5924d224ebd">
<img width="556" alt="image" src="https://github.com/moj-analytical-services/splink/assets/30687917/5f5c9d47-b5b5-4f04-9ead-83f9173aeeed">
Could you please help me, how can I solve the issue?
Thank you!

**Comment by RobinL** [comment]
Splink is really built more for situations where all the fields exhibit some fuzziness so it might not work particulalry well in this instance

Addresses are also notoriously difficult to deal with effectively.

So overall I think you'll probably struggle to get good results with your requirements.

I would probably recommend either doing the left join yourself, or doing a [determinstic link](https://moj-analytical-services.github.io/splink/demos/examples/duckdb/deterministic_dedupe.html?h=determi) on postal code and province, and then writing a custom sql query to choose addresses with a jaro winkler/jaccard/leven threshold above what you want.

You could do like
`df_predict = linker.deterministic_link()`
and then
df_predict.query_sql(your_query_here)
